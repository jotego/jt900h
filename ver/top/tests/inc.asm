    ; Arithmetic operations that read & write to memory
    main section code
    org 0
    ld a,0xbf    ; common header

    ld a,7
    inc 3,a
    cp a,10
    jp ne,bad_end
    or ra3,1

    inc 1,a
    cp a,11
    jp ne,bad_end
    or ra3,2

    lda xix,data
    ld b,(xix)
    inc 3,(xix)
    ld c,(xix)
    sub c,b
    cp c,3
    jp ne,bad_end
    or ra3,4

    ldf 3
    ldw (xix),0
    incw 3,(xix)
    ld bc,(xix)
    cp (xix),3
    jp ne,bad_end
    or ra3,8

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end