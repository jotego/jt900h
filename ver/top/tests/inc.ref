0000000b - 00000000 - 00000000 - 0000000f
0000fe03 - 00000000 - 00000000 - 00000003
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 0000babe
-- index registers --
00000066 - 00000000 - 00000000 - 00000100
SR = fb04
