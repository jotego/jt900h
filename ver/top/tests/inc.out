0000000b - 00000000 - 00000000 - 00000000
0000fe03 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
0000003f - 00000000 - 00000000 - 00000000
SR = 42f8
